### DISCUSSION TITLE
Rdkit runtime error: Pre condition Violation.

### DISCUSSION THREAD

**Discussion by sarahmarie935** [rdkit/rdkit#6475] [General]
Hi,

I have been having this error when I run my code but I am unsure how to approach to fix it.

RuntimeError: Pre-condition Violation
	bond already exists
	Violation occurred on line 334 in file Code/GraphMol/RWMol.cpp
	Failed Expression: !(boost::edge(atomIdx1, atomIdx2, d_graph).second)
	RDKIT: 2022.09.5
	BOOST: 1_78

Thank you,
Sarah

**Comment by jasondbiggs** [comment]
Please include the code which generates the message - a complete example that can be copy-pasted and run would be best.

**Reply by sarahmarie935** [reply]
I can add more put this is my input script  and beneath is the code in the .py file

/.prepare_dual_topology.py --config_file=step2.ini



 # Copy data
    [rdmol.SetProp(k, v) for k, v in pybel.MoleculeData(openbabel_obmol).items()]
    rdmol.SetProp('_Name', openbabel_obmol.GetTitle())

    for each_atom in rdmol.GetAtoms():
        if each_atom.GetBonds() != ():
            continue
        import numpy
        dist_list = numpy.argsort(numpy.array(rdkit.Chem.AllChem.Get3DDistanceMatrix(rdmol)[each_atom.GetIdx()]))
        try:
            closer_atom = int(dist_list[1])
        except IndexError:
            # Is this a lone atom?
            continue
        rdedmol = rdkit.Chem.RWMol(rdmol)
        rdedmol.AddBond(each_atom.GetIdx(), closer_atom)
        rdmol = rdedmol.GetMol()
        rdkit.Chem.SanitizeMol(rdmol)

        os_util.local_print('Atom id: {} is not explicitly bonded to any atom in molecule, connecting it to the closer '
                            'atom id: {}'.format(each_atom.GetIdx(), closer_atom),
                            current_verbosity=verbosity, msg_verbosity=os_util.verbosity_level.warning)

    rdkit.Chem.SanitizeMol(rdmol)
    rdkit.Chem.AssignStereochemistry(rdmol)
    rdkit.Chem.AssignStereochemistryFrom3D(rdmol)

**Reply by greglandrum** [reply]
Hi @sarahmarie935 the error message in the exception is telling you that you're trying to add a bond between two atoms which are already bonded to each other.
Without reading through your code carefully: could it be that it's trying to add each bond twice (once for atom A to atom B and again for atom B to atom A)?
An easy check would be:  `if rdedmol.GetBondBetweenAtoms(each_atom.GetIdx(),closer_atom) is None` to avoid adding duplicate bonds.
